contracts/eosiolib/multi_index.hpp

```diff
@@ -439,10 +439,18 @@ class multi_index
 
          auto size = db_get_i64( itr, nullptr, 0 );
          eosio_assert( size >= 0, "error reading iterator" );
-         char tmp[size];
-         db_get_i64( itr, tmp, uint32_t(size) );
 
-         datastream<const char*> ds(tmp,uint32_t(size));
+         char default_buffer[512];
```

Comment:
while this method should be on top of the stack instead of buried, it is still probably worth seeing if we can refactor the such that the "usual" case of a small item is still doing a tight stack allocation